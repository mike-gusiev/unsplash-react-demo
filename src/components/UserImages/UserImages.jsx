import React from 'react';
import PropTypes from 'prop-types';
import { Image, Row, Col } from 'react-bootstrap'

import './UserImages.scss';

export const UserImages = (props) => {
  const { images } = props;
  return (
    <div>
      <Row>
        {images.map((img, i) =>
          <Col xs={6} md={4} key={i}>
            <Image className="user-image" src={img} rounded/>
          </Col>)
        }
      </Row>
    </div>
  )
};

UserImages.propTypes = {
  images: PropTypes.array,
};
