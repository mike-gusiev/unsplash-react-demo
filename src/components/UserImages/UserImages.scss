.user-image {
  padding: 4px;
}