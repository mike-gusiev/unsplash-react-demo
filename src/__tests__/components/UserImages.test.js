import React from 'react';
import rendered from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { UserImages } from '../../components/UserImages/UserImages';

configure({adapter: new Adapter()});
const defaultProps = {
    images: [
      'url',
      'secondUrl'
    ]
};

describe('UserImages component', () => {

    test('matches the snapshot', () => {
        const tree = rendered.create(<Router><UserImages {...defaultProps}/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
