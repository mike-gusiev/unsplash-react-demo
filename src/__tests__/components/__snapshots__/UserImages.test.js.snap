// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserImages component matches the snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-4 col-xs-6"
    >
      <img
        className="user-image img-rounded"
        src="url"
      />
    </div>
    <div
      className="col-md-4 col-xs-6"
    >
      <img
        className="user-image img-rounded"
        src="secondUrl"
      />
    </div>
  </div>
</div>
`;
